libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/NavigationView.kt

```diff
@@ -109,7 +109,7 @@ class NavigationView @JvmOverloads constructor(
         keepScreenOn = true
         captureSystemBarsInsets()
 
-        SharedApp.setup(context.applicationContext as Application)
+        MapboxNavigationApp.registerObserver(SharedApp)
```

Comment:
I don't think we should be changing how `SharedApp` is integrated with `NavigationView`. Instead, we should leave `SharedApp.setup(Application)` call here and move `MapboxNavigationApp.registerObserver(SharedApp)` call inside `SharedApp.setup` method.
Registering `SharedApp` as `MapboxNavigationObserver` is an implementation detail that should be hidden from the public.

Reply:
Yeah am in agreement here. `SharedApp` is still internal, but what we will want is a public version of `SharedApp` and then it can register an internal `MapboxNavigationObserver` object

Reply:
Why should `setup(Application)` be required though? The application context is passed to `NavigationOptions`

Reply:
I added the `application context` requirement to get early access to the `Context` before `MapboxNavigation` and `NavigationOptions` are initialized.

Reply:
But why does `SharedApp` need the early access to the `Context`. Everything that uses the context would be better to get it from `MapboxNavigationObserver`